metakb/database.py

```diff
@@ -77,78 +77,108 @@ def load_from_json(self, infile_path: Path):
+        ]:
+            query = (
+                f"CREATE CONSTRAINT {label.lower()}_id_constraint "
+                f"IF NOT EXISTS ON (n:{label}) "
+                "ASSERT n.id IS UNIQUE;"
+            )
+            try:
+                tx.run(query)
+            except ServiceUnavailable as exception:
+                logging.error(f"Failed to generate ID property "
+                              f"constraint for {label}.")
+                raise exception
 
     def add_transformed_data(self, data: Dict):
         """Add set of data formatted per Common Data Model to DB.
         :param Dict data: contains key/value pairs for data objects to add
             to DB, including Assertions, Therapies, Diseases, Genes,
             Variations, Propositions, and Evidence
         """
+        added_ids = list()  # Used to keep track of IDs that are in statements
         with self.driver.session() as session:
             loaded_count = 0
+            for ev in data.get('statements', []):
+                self._get_ids_from_statement(ev, added_ids)
+            for var_descr in data.get('variation_descriptors', []):
+                if var_descr['id'] in added_ids:
+                    gc = var_descr['gene_context']
+                    if gc and gc not in added_ids:
+                        added_ids.append(gc)
             for method in data.get('methods', []):
-                session.write_transaction(self._add_method, method)
-            for descr in data.get('therapy_descriptors', []):
-                session.write_transaction(self._add_descriptor, descr)
-            for descr in data.get('disease_descriptors', []):
-                session.write_transaction(self._add_descriptor, descr)
-            for descr in data.get('gene_descriptors', []):
-                session.write_transaction(self._add_descriptor, descr)
+                try:
+                    session.write_transaction(self._add_method, method,
+                                              added_ids)
+                except ConstraintError:
+                    continue
+            for descriptor in ['therapy_descriptors', 'disease_descriptors',
+                               'gene_descriptors']:
+                for d in data.get(descriptor, []):
+                    try:
+                        session.write_transaction(
+                            self._add_descriptor, d, added_ids
+                        )
+                    except ConstraintError:
+                        continue
             for var_descr in data.get('variation_descriptors', []):
-                session.write_transaction(self._add_variation_descriptor,
-                                          var_descr)
+                try:
+                    session.write_transaction(self._add_variation_descriptor,
+                                              var_descr, added_ids)
+                except ConstraintError:
+                    continue
```

Comment:
Would be good to log any cases of this (+ for docs, props as well) if we aren't already